app/controllers/concerns/classified_listings_toolkit.rb

```diff
@@ -34,4 +36,166 @@ def bump_listing_success
   def clear_listings_cache
     ClassifiedListings::BustCacheJob.perform_now(@classified_listing.id)
   end
+
+  def set_classified_listing
+    @classified_listing = ClassifiedListing.find(params[:id])
+  end
+
+  def process_listing_creation
+    @classified_listing = ClassifiedListing.new(listing_params)
```

Comment:
This should be a service I guess :grimacing:  Though I don't suggest doing the refactoring in this pr, it should be a separate one. Personally, I would refactor to prepare for the feature and then implement it, but since the feature is implemented, if everything is working, probably it's worth to accept it and refactor later.